sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -660,6 +646,11 @@ struct _pi_context : _pi_object {
   // support of the multiple devices per context will be added.
   ze_command_list_handle_t ZeCommandListInit;
 
+  // Mutex for the immediate command list. Per the Level Zero spec memory copy
+  // operations submitted to an immediate command list are not allowed to be
+  // called from simultaneous threads.
+  std::mutex ImmediateCommandListMutex;
```

Comment:
don't we want a wrapper similar to `pi_shared_mutex` to avoid mutex overhead in single-thread mode?

Reply:
Fixed.